code/modules/client/client_procs.dm

```diff
@@ -236,6 +236,9 @@ var/global/list/localhost_addresses = list(
 	if(prefs && !istype(mob, world.mob))
 		prefs.apply_post_login_preferences()
 
+	if(SSinput.initialized)
+		set_macros()
```

Comment:
OK, yeah, didn't see it here before. It shouldn't be in all the places you put it in; use the char prefs thing to deal with it.

Reply:
Reduced places where it will call.

Reply:
We discussed this and apparently this needs to trigger each time a client's mob changes (?). In any case, that addresses this and related concerns.